### PR TITLE
feat(protocol): remove Taiko token dependency

### PR DESCRIPTION
**Pull request by AnshuJalan**
This PR removes the dependency of Taiko's L1 contract on Taiko token:
- All bonds are now paid in native ETH.
- ETH bonds need to be pre-deposited via existing `TaikoL1.depositBond` function. They can be withdrawn using `TaikoL1.withdrawBond`.
- `ProverSet` contract has been removed. It was used as a store of Taiko tokens and relayed block proving calls. It is no longer useful because the prover must deposit their ETH bonds by themselves in `TaikoL1`.
- Airdrop, Vesting and other team related contracts have been removed.
- Bond amounts have been changed to ETH values based on current TAIKO price. Eg: Liveness bond is changed from 125 TAIKO to 0.07 ETH.

### CODE REVIEW

**Review by smartprogrammer93 — APPROVED** [review]

File: packages/protocol/contracts/layer1/based/LibBonds.sol [hunk 1]
```diff
@@ -17,62 +13,45 @@ library LibBonds {
...
-        TaikoData.State storage _state,
-        IAddressResolver _resolver,
-        uint256 _amount
-    )
-        internal
-    {
+    function depositBond(TaikoData.State storage _state, uint256 _amount) internal {
         _state.bondBalance[msg.sender] += _amount;
-        _tko(_resolver).transferFrom(msg.sender, address(this), _amount);
     }
```
[the remark this answers is not in the document]
Comment (AnshuJalan) [reply packages/protocol/contracts/layer1/based/LibBonds.sol:24]: It is crediting the entire amount of ETH that is sent along the external `depositBond` call in `TaikoL1`: https://github.com/NethermindEth/taiko-mono-core-rollup/pull/2/files#diff-108b8624155bf1271d820a38ab19ef7cb9b30d91cd199b8ae149fbc0e81a1193R168.

File: packages/protocol/contracts/layer1/based/TaikoL1.sol [hunk 2]
```diff
@@ -295,7 +295,7 @@ contract TaikoL1 is EssentialContract, ITaikoL1, TaikoEvents {
             blockRingBufferSize: 360_000, // = 7200 * 50
             maxBlocksToVerify: 16,
             blockMaxGasLimit: 240_000_000,
-            livenessBond: 125e18, // 125 Taiko token
+            livenessBond: 0.07 ether,
```
Comment (e1Ru1o) [inline packages/protocol/contracts/layer1/based/TaikoL1.sol:298]: any relevant comment about why this number is used

Comment (AnshuJalan) [reply packages/protocol/contracts/layer1/based/TaikoL1.sol:298]: Last point in PR description (checked with Lin from the research team):
> Bond amounts have been changed to ETH values based on current TAIKO price. Eg: Liveness bond is changed from 125 TAIKO to 0.07 ETH.

File: packages/protocol/contracts/layer1/tiers/TierProviderBase.sol [hunk 3]
```diff
@@ -14,7 +14,7 @@ abstract contract TierProviderBase is ITierProvider {
     /// asynchronous, this grace period allows provers to defer submissions until gas
     /// prices become more favorable, potentially reducing transaction costs.
     uint16 public constant GRACE_PERIOD = 240; // minutes
-    uint96 public constant BOND_UNIT = 75 ether; // TAIKO tokens
```
Comment (e1Ru1o) [inline packages/protocol/contracts/layer1/tiers/TierProviderBase.sol:17]: double checking numbers as well

File: packages/protocol/contracts/layer1/tiers/TierProviderBase.sol [hunk 4]
```diff
@@ -14,7 +14,7 @@ abstract contract TierProviderBase is ITierProvider {
...
+    uint96 public constant BOND_UNIT = 0.04 ether;
```
Comment (e1Ru1o) [inline packages/protocol/contracts/layer1/tiers/TierProviderBase.sol:17]: double checking numbers here too

File: packages/protocol/script/layer1/deploy_protocol_on_l1.sh [hunk 5]
```diff
@@ -3,44 +3,26 @@
 # This script is only used by `pnpm test:deploy:l1`.
 set -e
 
-export PRIVATE_KEY=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
+export PRIVATE_KEY=0xbcdf20249abf0ed6d944c0288fad489e33f66b3960d9e6229c1cd214ed3bbe31
```
Comment (e1Ru1o) [inline packages/protocol/script/layer1/deploy_protocol_on_l1.sh:6]: what is this private key, and why it makes sense to have it in plain text here. I see taiko had one as well.

Comment (AnshuJalan) [reply packages/protocol/script/layer1/deploy_protocol_on_l1.sh:6]: This is a test private key on kurtosis devnet. This script is only going to be used in test deployment within Kurtosis package.

File: packages/protocol/test/layer1/based/TaikoL1TestBase.sol [hunk 6]
```diff
@@ -255,14 +244,11 @@ abstract contract TaikoL1TestBase is TaikoTest {
...
     }
 
-    function giveEthAndTko(address to, uint256 amountTko, uint256 amountEth) internal {
-        vm.deal(to, amountEth);
-        tko.transfer(to, amountTko);
-
-        vm.prank(to, to);
-        tko.approve(address(L1), amountTko);
+    function giveEthAndDepositBond(address to, uint256 amountEth) internal {
+        vm.deal(to, amountEth + 1000 ether); // Extra eth compensates for block proposals
```
Comment (e1Ru1o) [inline packages/protocol/test/layer1/based/TaikoL1TestBase.sol:248]: this extra amount granted here breaks a bit the logic of the function, if I write tests and use it I would expect only the `amountEth` to be granted. Notice that if not I will need to keep it in mind for balance checks, and if you decide to change the number granted here all my tests will break.

Comment (AnshuJalan) [reply packages/protocol/test/layer1/based/TaikoL1TestBase.sol:248]: I agree. It wouldn't be fair to be done this way when writing from scratch, however, this is done here to keep it in line with the former tests of Taiko, where besides an arbitrary taiko token amount (10,000 TKO tokens), additional 1000 eth was sent to process block proposals: https://github.com/NethermindEth/taiko-mono-core-rollup/pull/2/files/4dfc2ec79c44101a0cc457b0fa23462b9028c7c4#diff-e92c7ce15871475760272c5000c1e390acb53eea8c383f38ddc8d7dd092f9a70L15

**Review by e1Ru1o — COMMENTED** [review]
Please double check changes in `packages/taiko-client/bindings/gen*`, `packages/protocol/test/genesis/data/genesis*` and `packages/taiko-client/bindings/.githead` as I skipped them because I lack the context

**Comment by AnshuJalan** [comment]
> Please double check changes in `packages/taiko-client/bindings/gen*`, `packages/protocol/test/genesis/data/genesis*` and `packages/taiko-client/bindings/.githead` as I skipped them because I lack the context

Will be done by @mikhailUshakoff

File: packages/protocol/contracts/layer1/tiers/TierProviderBase.sol (hunk 4 above)
Comment (AnshuJalan) [reply packages/protocol/contracts/layer1/tiers/TierProviderBase.sol:17]: Same interpretation as the previous value of 0.07 eth. Based on Taiko price.

File: packages/protocol/test/layer1/based/TaikoL1TestBase.sol (hunk 6 above)
Comment (e1Ru1o) [reply packages/protocol/test/layer1/based/TaikoL1TestBase.sol:248]: can you keep it like the original function just for it to be more extensible and more friendly if future tests are needed. Kind of:
```solidity
function giveEthAndDepositBond(address to, uint256 bondEth, uint256 proposalEth) internal {
        vm.deal(to, amountEth + proposalEth);
```
That way you can control the amount in the tests

Comment (AnshuJalan) [reply packages/protocol/test/layer1/based/TaikoL1TestBase.sol]: Fair. Just pushed this change

**Comment by mskrzypkows** [comment]
`gen_*` files are generated using `./scripts/gen_bindings.sh` they are bindings to the changed contracts.

**Review by mikhailUshakoff — APPROVED** [review]

**Pull request merged**